src/utils/check-utils.ts

```diff
@@ -0,0 +1,62 @@
+import { SEMVER_LABELS } from '../constants';
+import { CheckRunStatus, CheckStatus, PRContext } from '../types';
+import { getSemverLabel } from './label-utils';
+
+// Define the possible check statuses
+export const checkStatuses = {
+  validNoChanges: {
+    conclusion: CheckRunStatus.SUCCESS,
+    title: 'No Changes',
+    summary: "We couldn't see any changes in the `electron.d.ts` artifact",
+  },
+  invalidNoChanges: {
+    conclusion: CheckRunStatus.FAILURE,
+    title: 'Label Mismatch with No Changes',
+    summary: "No changes detected despite the presence of 'semver/minor' or 'semver/major' labels.",
```

Comment:
There needs to be a way of overriding this, something like like our `fast-track` or `skip-backport-check` labels to bypass trop / cation checks.

This isn't 100% provably correct, for instance:
* You can make a `semver/major` change with 0 API surface changing, for instance enabling a feature by default (Cooke Encryption comes to mind as a semver/major with 0 API)
* You can make a `semver/none` change that modifies the API surface.  For instance docs fixes commonly "modify" the `electron.d.ts` file but don't actually _change_ the API surface. So unless this check gets smarter around diffing actual API surface rather than just the `electron.d.ts` file we need a way to say "in this case we're all good"

Reply:
Thanks for pointing out the edge cases that I haven't considered!

I can add a new label that's `skip-type-diff-check` but do you think it adds unnecessary complexity to the process? After all, the primary purpose of these checks is to provide informational cues to users.

What if for expected `valid` changes, we set the status to `CheckRunStatus.SUCCESS` and for suspected `invalid` changes, we set the status to `CheckRunStatus.NEUTRAL` with a summary that gives a warning that unexpected changes exist but feel free to skip if it's intentional.

This way, it serves the purpose of providing information without adding process overhead. If we feel like the override tag is better, happy to add it!!